### DISCUSSION TITLE
I forgot wallet password

### DISCUSSION THREAD

**Discussion by Gugg-Z** [lightningnetwork/lnd#8114] [Q&A] [answered]
[lncli] rpc error: code = Unknown desc = invalid passphrase for master public key
I forgot wallet password, how can I retrieve it?
I only saved mnemonic phrase.

**Comment by guggero** [comment, marked as answer]
You can't, otherwise the password wouldn't be secure after all. What you can do is re-create the wallet from the seed:
1. Stop `lnd`. Rename `~/.lnd/data/chain/bitcoin/mainnet/wallet.db` to `wallet.db.backup` (or something like that).
2. Delete `~/.lnd/data/chain/bitcoin/mainnet/macaroons.db` and all `*.macaroon` files in `~/.lnd/data/chain/bitcoin/mainnet/`.
3. DO NOT DELETE OR MOVE ANY OTHER FILES!
4. Start `lnd`.
5. Run `lncli wallet create` and enter the mnemonic phrase.
6. This will re-scan the chain and can take a couple of hours, don't interrupt the process and watch the log for progress.
7. Once the wallet is re-scanned, all channels should come online again normally.

**Reply by Gugg-Z** [reply]
Thank you!
Will the above operations influence old wallet balance?

**Reply by guggero** [reply]
It should restore the wallet balance, yes. Just make sure you use the correct seed that you used before. Otherwise if you had active channels and use a different seed, very weird errors will probably be thrown.

**Reply by Gugg-Z** [reply]
![1698243464849](https://github.com/lightningnetwork/lnd/assets/28720689/2cd991cd-e147-486f-8ddd-fb0b39bd840b)
![1698243357424](https://github.com/lightningnetwork/lnd/assets/28720689/72499aff-d50e-4dc5-a7a5-b6545359512b)
![image](https://github.com/lightningnetwork/lnd/assets/28720689/30ce59ce-c940-486e-8d14-f1215e537ccb)
I cant find wallet.

**Reply by guggero** [reply]
Did you use a passphrase for the mnemonic seed when you created that? Looks like the wallet wasn't created because the passphrase was incorrect.

**Reply by Gugg-Z** [reply]
![image](https://github.com/lightningnetwork/lnd/assets/28720689/0df6dc0a-142a-41f6-a39a-84f2aff10c26)
So should I use cipher seed passphrase?
I didnt use cipher seed passphrase when I created old wallet.

**Reply by guggero** [reply]
Yes, the cipher seed passphrase. Well, either you entered the seed phrase incorrectly or you **did** use cipher seed passphrase when you created the wallet.

**Reply by Gugg-Z** [reply]
Many thanks for your kind and warm help.
I restore my wallet.
